src/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import cx from '../utils/classnames';
+import { props, t, ReactChildren } from '../utils';
+import FlexView from 'react-flexview';
+
+export type NavBarProps = {
+  /** Left content: It doesn't shrink nor grow */
+  /** Center content. It grows to use all the available space */
+  /** Right content. It doesn't shrink nor grow */
+  /** Max Width. For css `max-width` */
+  content: {
+    left?: any, // TODO: ReactChildren
+    center?: any, // TODO: t.ReactChildren
+    right?: any, // TODO: t.ReactChildren,
+    maxWidth?: string | number
+  },
+  /** to set `position: fixed` */
+  fixed?: boolean,
+  /** shorthand for css `height` */
+  height?: string | number,
+  /** shorthand for css `background` */
+  background?: string,
+  /** add class name */
+  className?: string,
+  /** add custom css style */
+  style?: React.CSSProperties
+}
+
+export const Props = {
+  content: t.struct({
+    left: t.maybe(ReactChildren),
+    center: t.maybe(ReactChildren),
+    right: t.maybe(ReactChildren),
+    maxWidth: t.maybe(t.union([t.String, t.Number]))
+  }),
+  fixed: t.maybe(t.Boolean),
+  height: t.maybe(t.union([t.String, t.Number])),
+  background: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+};
+
+@props(Props)
+export default class NavBar extends React.PureComponent<NavBarProps> {
```

Comment:
It wasn't a `PureComponent` before

Reply:
@francescogior he's right, but we've already changed most of them during the other refactors so I would leave it like this (we're going to kill `@pure` anyway before a release)